mbuild/packing.py

```diff
@@ -354,6 +358,23 @@ def fill_region(
                 "`compound`, `n_compounds`, and `fix_orientation` must be of "
                 "equal length."
             )
+    if bounds is not None:
+        if not isinstance(bounds, (list)):
+            raise TypeError(
+                "`bounds` must be a list of one or more bounding boxes "
+                "and/or `None` for each item in `compound`."
+            )
+        if len(bounds) != len(n_compounds):
+            raise ValueError(
+                "`compound` and `bounds` must be of equal length. Use `None` "
+                "for non-bounded items in `compound`."
+            )
+        for bound in bounds:
+            if not isinstance(bound, (Box, list)):
+                raise ValueError(
```

Comment:
A unit test that this error is called would be great in test_packing.py. Same for the error on line 368.

Reply:
Second this.
An example of how to invoke an error for testing with pytest:
```
    def test_n_bonds_particle(self):
        comp = mb.Compound(name="A", pos=[0, 0, 0])
        with pytest.raises(MBuildError):
            comp.n_bonds
```